server/tests/e2e/decentralized.spec.ts

```diff
@@ -114,4 +119,148 @@ describe('end-to-end decentralized', function () {
   it('several rounds of cifar 10 with three secure aggregators yields consensus', async () => {
     await reachConsensus('secure', 3)
   })
+
+  it("peers emit expected statuses", async function () {
+    this.timeout(15_000);
+    const lusCovidTask = defaultTasks.lusCovid.getTask();
+    lusCovidTask.trainingInformation.aggregationStrategy = 'mean';
+    lusCovidTask.trainingInformation.epochs = 8;
+    lusCovidTask.trainingInformation.roundDuration = 2;
+    lusCovidTask.trainingInformation.minNbOfParticipants = 2;
+
+    const DATASET_DIR = path.join("..", "datasets");
+
+    const [positive, negative] = [
+      (
+        await loadImagesInDir(path.join(DATASET_DIR, "lus_covid", "COVID+"))
+      ).zip(Repeat("COVID-Positive")),
+      (
+        await loadImagesInDir(path.join(DATASET_DIR, "lus_covid", "COVID-"))
+      ).zip(Repeat("COVID-Negative")),
+    ];
+    const dataset = positive.chain(negative);
+
+    /**
+     * Then at each round (each call to `disco.trainByRound`) the event cycle is:
+     * a) During onRoundBeingCommunication, 
+     *   1. the peer notifies the server that they want to join the next round
+     *   2. finishes by updating the status to TRAINING
+     * (without waiting for a server answer)
+     * b) local training (the status remains TRAINING)
+     * c) During onRoundEndCommunication 
+     *   1. the peer notifies the server that they are ready to share weights 
+     *      set status to RETRIEVING PEERS
+     *   2. wait for the server to answer with the current round's peers list
+     *      this is where the nb of participants is updated
+     *   3. establish peer-to-peer connections 
+     *   4. set status to UPDATING MODEL and exchange weight updates
+     * 
+     * Given this, it is important to note that calling disco.trainByRound().next()
+     * for the first time will perform a) and then b) where it stops and yields the round logs.
+     * Thus, c) isn't called and the weight sharing is not performed during this call to next().
+     * Calling next() again will then run c), as well as a) and b) again.
+     * 
+     * In this test the timeline is:
+     * - User 1 joins the task by themselves
+     * - User 2 joins
+     * - User 1 leaves
+     * - User 3 joins
+     * - User 2 & 3 leave
+     */
+    const statusUpdateTime = 500 // allow some time for the client to update their status
```

Comment:
yeah, time based tests are broken; using [an event queue](https://github.com/epfml/disco/blob/88b20b34b14d9cecf196e081d984c6a2e4413d65/server/tests/e2e/federated.spec.ts#L26) (from #781) to check value changes is safer and faster. you can ensure that a status didn't "flicker" and that it hadn't gone through unexpected status before stabilisation.
I actually did it in there as the tests were very regularly failing on my laptop.